### ISSUE TITLE
Negative amount shows in workspace invoice room on fresh sign-in

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#47000]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.17-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @jjcoffee
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1723024665726109

## Action Performed:
Precondition: Having User A as a high traffic account with multiple workspaces
1. [User A] Send an invoice to any workspace.
2. [User A] Copy the link to the invoice room.
3. [User B] Open the invoice, tap Pay and select to pay the invoice as an individual and pay elsewhere.
4. [User A] Sign out and sign back in.
5. [User A] Paste and open the link in (2)

## Expected Result:
The invoice room opens with a report preview showing the correct payment amount, avatars and names showing correctly.
## Actual Result:
The payment amount shows as negative and states "owes" even though it has been paid. The avatar and names are also blank.
## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos
![image (14)](https://github.com/user-attachments/assets/b76c1905-695f-455e-bdcf-73df41164d4c)

https://github.com/user-attachments/assets/9f86ff6b-e16a-4250-92dc-d4cd13486c13

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by jliexpensify** [comment]
Hi @rayane-djouah are you able to reproduce this one?

There's a precondition: Having User A as a high traffic account with multiple workspaces (I don't have a high-traffic account)

**Comment by NJ-2020** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Negative amount shows in workspace invoice room on fresh sign-in
### What is the root cause of that problem?
We're getting the value from `iouReport` for the invoice report, but after we fresh sign in and open the room, we did not yet fetch the Invoice report data, so when we try to find inside the Onyx data it will return undefined
https://github.com/Expensify/App/blob/87bdc246318f6c3fc81ad9d293024de687e457a2/src/components/ReportActionItem/ReportPreview.tsx#L529-L531
So when we open the Invoice report and go back it will show the correct info because we've fetched the invoice report data
### What changes do you think we should make in order to solve the problem?
We can check if the `iouReport` data is undefined, then we can fetch the Invoice report by using `openReport` function
```ts
const fetchIOUReport = useCallback(() => {
    ReportActions.openReport(iouReportID);
}, [iouReportID]);

useEffect(() => {
    if(!isEmptyObject(iouReport)) {
        return;
    }

    fetchIOUReport()
}, [iouReport])
```
### What alternative solutions did you explore? (Optional)

**Comment by NJ-2020** [comment]
@rayane-djouah Please kindly review above proposal. Thanks

**Comment by NJ-2020** [comment]
@rayane-djouah Yes its not happening with expense reports, I think this is because BE issue like for negative amount the BE return text with negative amount on the `message` property:

![Screenshot 2024-08-13 at 22 53 04](https://github.com/user-attachments/assets/8a0c53f6-b28d-495e-829b-4a9963a53282)

For avatar returning undefined because the `ownerAccountID` & `actorAccountID` return undefined because of backend issue, but when checking on the action report the BE return `actorAccountID` with a value
![Screenshot 2024-08-13 at 22 57 51](https://github.com/user-attachments/assets/89060c68-43d1-43c1-87db-6c2d421abfa0)
https://github.com/Expensify/App/blob/b7f7f1b4f36451b4e3a763ac717e1c8fd645ee5a/src/pages/home/report/ReportActionItemSingle.tsx#L84
https://github.com/Expensify/App/blob/b7f7f1b4f36451b4e3a763ac717e1c8fd645ee5a/src/pages/home/report/ReportActionItemSingle.tsx#L96

But here we're getting the property value of `managerID` or `ownerAccountID ` inside of iouReport(undefined because not yet fetched) or `childManagerAccountID` or `childOwnerAccountID` inside reportAction
https://github.com/Expensify/App/blob/b7f7f1b4f36451b4e3a763ac717e1c8fd645ee5a/src/libs/ReportUtils.ts#L7358-L7361

And also for text `owes` instead of paid because on the iouSettled we're checking inside iouReport or `reportAction.childStatusNum` is equal to `CONST.REPORT.STATUS_NUM.REIMBURSED` which is 4, but the `reportAction.childStatusNum` returning 0 which will return false and return the `owes` text instead of paid


But i think we can solve it on the FE by fetching the Invoice report data
Thanks

**Comment by NJ-2020** [comment]
@rayane-djouah Please let me know your thought.

**Comment by cristipaval** [comment]
I need to check how it works for the expense reports.

**Comment by jliexpensify** [comment]
Not overdue, waiting on Cristi

**Comment by cristipaval** [comment]
On my list for today.

**Comment by jliexpensify** [comment]
Waiting on @cristipaval

**Comment by NJ-2020** [comment]
Bump @cristipaval ^

**Comment by jliexpensify** [comment]
DM-ed Cristi

**Comment by cristipaval** [comment]
I can't reproduce this one with the latest develop.

**Comment by NJ-2020** [comment]
@cristipaval The issue is still reproducible on the latest main, note: you must have a high traffic account
![Screenshot 2024-08-26 at 20 20 48](https://github.com/user-attachments/assets/09b79298-3a91-4d05-be3d-aa63ddabcedc)

**Comment by jliexpensify** [comment]
Bump @cristipaval

**Comment by cristipaval** [comment]
Unfortunately, I won't have time to dig more into this one, and I'm off starting on Monday. Maybe @madmax330 could help, given that he also has knowledge about Invoicing.

**Comment by techievivek** [comment]
Not overdue, will review it tomorrow, thanks.

**Comment by madmax330** [comment]
What do we do for other expenses? Do we pass the negative amount from the backend or is it a positive amount?

**Comment by NJ-2020** [comment]
@madmax330 Can you check this [[comment]()](https://github.com/Expensify/App/issues/47000#issuecomment-2287928656)

**Comment by jliexpensify** [comment]
@madmax330 to confirm, this should now be `Internal`?

**Comment by madmax330** [comment]
Yeah

**Comment by jliexpensify** [comment]
Done, feel free to adjust the priority

**Comment by techievivek** [comment]
Not overdue, @madmax330 will you be taking over this since you seem to have context on the changes that might be required here? 🙏

**Comment by madmax330** [comment]
Sure

**Comment by jliexpensify** [comment]
@madmax330 can we get a sense of priority here? Should we drop to `Monthly` or `weekly` since Billpay isn't being prioritised?

**Comment by madmax330** [comment]
Changed it to weekly for now

**Comment by madmax330** [comment]
Not a priority yet

**Comment by jliexpensify** [comment]
Bump @madmax330 to address [this](https://github.com/Expensify/App/issues/47000#issuecomment-2402861867)

**Comment by madmax330** [comment]
I'll look into it next week. I'm OOO starting tomorrow

**Comment by madmax330** [comment]
Haven't had time for this, not really a priority

**Comment by jliexpensify** [comment]
Waiting on Max

**Comment by madmax330** [comment]
Will look into this when I'm back from OOO

**Comment by madmax330** [comment]
Not really a priority

**Comment by jliexpensify** [comment]
[Posted](https://expensify.slack.com/archives/C9YU7BX5M/p1740091975554439)

**Comment by m-natarajan** [comment]
Still able to reproduce.

https://github.com/user-attachments/assets/f15ac58d-3ccf-46ce-9763-8318f956c848

**Comment by jliexpensify** [comment]
This is an Internal issue so @madmax330 will have the best context on Next Steps.

Max - are we deprioritising BillPay issues? If so, do you feel like this needs to be worked on or closed?

**Comment by jliexpensify** [comment]
@madmax330 shall we close this?
